### DISCUSSION TITLE
Spancat single label validation error-extra fields not permitted

### DISCUSSION THREAD

**Discussion by Yl-C** [explosion/spaCy#12858] [Help: Coding & Implementations] [answered]
Using the first example in the [documentation for the span categorizer ](https://spacy.io/api/spancategorizer)
The following code mostly from the example
```python
import spacy
from spacy.pipeline.spancat import DEFAULT_SPANCAT_SINGLELABEL_MODEL


config = {
    "threshold": 0.5,
    "spans_key": "labeled_spans",
    "model": DEFAULT_SPANCAT_SINGLELABEL_MODEL,
    "suggester": {"@misc": "spacy.ngram_suggester.v1", "sizes": [1, 2, 3]},
    # Additional spancat_singlelabel parameters
    "negative_weight": 0.8,
    "allow_overlap": True,
}

nlp = spacy.blank("en")
nlp.add_pipe("spancat_singlelabel", config=config)
```
gives the following error
`Config validation error
spancat_singlelabel -> threshold	extra fields not permitted`

If I remove the threshold in the config, the spancat won't initialize.
I can't figure out what am I missing here.

**Comment by shadeMe** [comment, marked as answer]
Hi, thanks for bringing this to our attention! The example code is incorrect - the single label SpanCategorizer doesn't take a `threshold` argument. We'll fix that in the next release.

That being said, the example code is just a snippet that constructs and adds the SpanCat component to the pipeline. If you wish to train a pipeline with it, the setup is a bit more involved - the [demo project](https://github.com/explosion/projects/tree/v3/pipelines/spancat_demo) should provide a comprehensive overview of how this is done.

**Reply by KR1470R** [reply]
i'm trying to use this config too, but i 'm getting this error.
here's full output:
Config validation error
```
ner -> allow_overlap    extra fields not permitted
ner -> negative_weight  extra fields not permitted
ner -> spans_key        extra fields not permitted
ner -> suggester        extra fields not permitted
ner -> threshold        extra fields not permitted
{'nlp': <spacy.lang.en.English object at 0x7fef258dfc10>, 'name': 'ner', 'allow_overlap': True, 'incorrect_spans_key': None, 'model': {'@architectures': 'spacy.SpanCategorizer.v1', 'scorer': {'@layers': 'Softmax.v2'}, 'reducer': {'@layers': 'spacy.mean_max_reducer.v1', 'hidden_size': 128}, 'tok2vec': {'@architectures': 'spacy.Tok2Vec.v2', 'embed': {'@architectures': 'spacy.MultiHashEmbed.v1', 'width': 96, 'rows': [5000, 1000, 2500, 1000], 'attrs': ['NORM', 'PREFIX', 'SUFFIX', 'SHAPE'], 'include_static_vectors': False}, 'encode': {'@architectures': 'spacy.MaxoutWindowEncoder.v2', 'width': 96, 'window_size': 1, 'maxout_pieces': 3, 'depth': 4}}}, 'moves': None, 'negative_weight': 0.8, 'scorer': {'@scorers': 'spacy.ner_scorer.v1'}, 'spans_key': 'labeled_spans', 'suggester': {'@misc': 'spacy.ngram_suggester.v1', 'sizes': [1, 2, 3]}, 'threshold': 0.5, 'update_with_oracle_cut_size': 100, '@factories': 'ner'}

```

**Reply by svlandeg** [reply]
@KR1470R: this looks like a different issue - it seems like your config is mixing settings from `ner` and `spancat`. Can you open a new issue, paste your config file and explain how you got to that config file and what you're trying to do? Then we can help you better.

**Reply by KR1470R** [reply]
hi, thanks for reply, i've fixed that already, the problem was that i ran my model with incompatible spacy version.
